tests fail when using ./all.bash

When attempting to install zoekt, many tests fail. Is this expected?

One example below. Checked out at https://github.com/sourcegraph/zoekt/commit/5ddb944a6d3c5de8689da56592687afc3fa26e07

```
2023/03/24 04:23:43 loading 1 shard(s): repo_v16.00000.zoekt                                                                                                                                                                                                                    
--- FAIL: TestSkipSubmodules (0.57s)                                                                                                                                                                                                                                            
    tree_test.go:466: createMultibranchRepo: execution error: exit status 128, output + mkdir adir bdir                                                                                                                                                                         
        + cd adir                                                                                                                                                                                                                                                               
        + git init -b master                                                                                                                                                                                                                                                    
        Initialized empty Git repository in /usr/local/google/tmp/TestSkipSubmodules132999780/001/adir/.git/                                                                                                                                                                    
        + mkdir subdir                                                                                                                  
        + echo acont                                                                                                                                                                                                                                                            
        + echo sub-cont                                                                                                                 
        + git add afile subdir/sub-file                                                                                                                                                                                                                                         
        + git config user.email you@example.com                                                                                         
        + git config user.name 'Your Name'                                                                                                                                                                                                                                      
        + git commit -am amsg                                                                                                           
        [master (root-commit) f9c87f6] amsg                                                                                                                                                                                                                                     
         2 files changed, 2 insertions(+)                                                                                                                                                                                                                                       
         create mode 100644 afile                                                                                                                                                                                                                                               
         create mode 100644 subdir/sub-file                                                                                                                                                                                                                                     
        + cd ..                                                                                                                         
        + cd bdir                                                                                                                                                                                                                                                               
        + git init -b master                                                                                                            
        Initialized empty Git repository in /usr/local/google/tmp/TestSkipSubmodules132999780/001/bdir/.git/                                                                                                                                                                    
        + echo bcont                                                                                                                                                                                                                                                            
        + ln -s bfile bsymlink                                                                                                                                                                                                                                                  
        + git add bfile bsymlink                                                                                                                                                                                                                                                
        + git config user.email you@example.com                                                                                                                                                                                                                                 
        + git config user.name 'Your Name'                                                                                              
        + git commit -am bmsg                                                                                                           
        [master (root-commit) 63a7b2b] bmsg                                                                                                                                                                                                                                     
         2 files changed, 2 insertions(+)                                                                                               
         create mode 100644 bfile                                                                                                                                                                                                                                               
         create mode 120000 bsymlink                              
        + cd ../adir                                                                                                                                                                                                                                                            
        + git submodule add --name bname -- ../bdir bname                                                                               
        Cloning into '/usr/local/google/tmp/TestSkipSubmodules132999780/001/adir/bname'...                                                                                                                                                                                      
        fatal: transport 'file' not allowed                                                                                                                                                                                                                                     
        fatal: clone of '/usr/local/google/tmp/TestSkipSubmodules132999780/001/bdir' into submodule path '/usr/local/google/tmp/TestSkipSubmodules132999780/001/adir/bname' failed                                                                                              
2023/03/24 04:23:44 finished /usr/local/google/tmp/TestFullAndShortRefNames3149365922/002/repo_v16.00000.zoekt: 1966 index bytes (overhead 37.1)                                                                                                                                
2023/03/24 04:23:44 loading 1 shard(s): repo_v16.00000.zoekt                                                                                                                                                                                                                    
2023/03/24 04:23:44 finished /usr/local/google/tmp/TestLatestCommit3747486278/002/repo_v16.00000.zoekt: 1871 index bytes (overhead 35.3)                                                                                                                                        
2023/03/24 04:23:44 loading 1 shard(s): repo_v16.00000.zoekt                                                                            
FAIL                                                                                                            
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,9 +15,6 @@ jobs:
         run: apk add go git
       - name: install ctags
         run: ./install-ctags-alpine.sh
-      # Needed for submodule tests in gitindex
-      - name: git protocol.file.allow
-        run: git config --global --replace-all protocol.file.allow always
       - name: test
         run: go test ./...
 

--- a/gitindex/tree_test.go
+++ b/gitindex/tree_test.go
@@ -40,7 +40,13 @@ func createSubmoduleRepo(dir string) error {
 	if err := os.MkdirAll(dir, 0o755); err != nil {
 		return err
 	}
-	script := `mkdir adir bdir
+	script := `
+# Fix fatal: transport 'file' not allowed
+export GIT_CONFIG_COUNT=1
+export GIT_CONFIG_KEY_0=protocol.file.allow
+export GIT_CONFIG_VALUE_0=always
+
+mkdir adir bdir
 cd adir
 git init -b master
 mkdir subdir

